app/models/user.rb

```diff
@@ -0,0 +1,14 @@
+class User < ActiveRecord::Base
+  has_many :tweets
+  has_secure_password
+  validates_presence_of :username, :email
+
+  def slug
+    self.username.downcase.gsub(" ", "-")
```

Comment:
This is interesting...wouldn't it be possible to have multiple users with the same username and/or slug? For example '@john smith' and '@john-smith'.